config/src/main/java/org/axonframework/config/Configurer.java

```diff
@@ -476,6 +477,24 @@ default <A> Configurer configureAggregate(@Nonnull Class<A> aggregate) {
     Configurer registerHandlerDefinition(
             @Nonnull BiFunction<Configuration, Class, HandlerDefinition> handlerDefinitionClass);
 
+    /**
+     * Registers a builder function for a {@link HandlerEnhancerDefinition} used during constructing of the default
+     * {@link HandlerDefinition}.
+     * <p>
+     * Any number of handler enhancer builder functions can be registered through this method. Note that any
+     * {@code HandlerEnhancerDefinitions} registered through this method are <b>ignored</b> for handlers matching the
+     * type used in the {@link #registerHandlerDefinition(BiFunction)} method's lambda.
+     *
+     * @param handlerEnhancerBuilder A lambda constructing a {@link HandlerEnhancerDefinition} based on the
+     *                               {@link Configuration}.
+     * @return The current instance of the {@link Configurer}, for chaining purposes.
+     */
+    default Configurer registerHandlerEnhancerDefinition(
+            Function<Configuration, HandlerEnhancerDefinition> handlerEnhancerBuilder
+    ) {
+        return this;
```

Comment:
Given this is an interface we don't expect people to implement, but do expect people to use, I would refrain from providing a no-op default implementation. Either throw a "UnsupportedOperationException" or just leave the method abstract to cause a compiler error.
Since we are expected only to implement this ourselves, I would opt for the abstract method approach.

Reply:
Fair pointer. I'll remove the `default` implementation entirely.